operatoringress/v1/types.go

```diff
@@ -55,6 +55,14 @@ type DNSRecordSpec struct {
 	// +kubebuilder:validation:Minimum=0
 	// +required
 	RecordTTL int64 `json:"recordTTL"`
+	// dnsManagementPolicy denotes the current policy applied on the dns record.
+	// Records that have policy set as "Unmanaged" are ignored by the ingress
+	// operator and should be managed at the discretion of the cluster admin.
+	//
+	// +kubebuilder:default:="Managed"
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=Managed;Unmanaged
+	DNSManagementPolicy DNSManagementPolicy `json:"dnsManagementPolicy"`
```

Comment:
Why would we have a DNSRecord resource in cases where we do not managed the DNS?

Reply:
New created ingresscontrollers with `dnsManagementPolicy` set to `Unmanaged` will not create a *DNSRecord*, but in scenarios where we update the IC from `Managed` (current default) to `Unmanaged` we will have an associated *DNSRecord* with `dnsManagementPolicy` set to `Unmanaged`. The cluster admin can later on clean up the resource as needed.

Reply:
@thejasn just to want to be clear, a new IC will indeed have a dnsRecord, right? This might be an update since you wrote this comment.

Reply:
Yes, this was clarified ([here](https://github.com/openshift/api/pull/1200#issuecomment-1144896101)).